Sources/NIOPosix/SocketChannel.swift

```diff
@@ -508,7 +508,14 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
                 // Receiving packet info is only supported for IP
                 throw ChannelError.operationUnsupported
             }
-            break
+        case _ as ChannelOptions.Types.DatagramGenericSegmentationOffload:
+            guard System.supportsUDPSegmentationOffload else {
+                throw ChannelError.operationUnsupported
+            }
+            let segmentSize = value as! ChannelOptions.Types.DatagramGenericSegmentationOffload.Value
+            try self.socket.withUnsafeHandle {
```

Comment:
Can we make this a `func` on the socket type itself? That way we can avoid having the unsafe code in this location, which would be nice.